group_project_v2/mixins.py

```diff
@@ -125,7 +126,180 @@ def real_user_id(self, anonymous_student_id):
         return self._known_real_user_ids[anonymous_student_id]
 
 
-class WorkgroupAwareXBlockMixin(UserAwareXBlockMixin, CourseAwareXBlockMixin):
+class SettingsMixin(object):
+
+    def _get_setting(self, setting, default):
+        result = default
+        settings_service = self.runtime.service(self, "settings")
+        if settings_service:
+            xblock_settings = settings_service.get_settings_bucket(self)
+            if xblock_settings and setting in xblock_settings:
+                result = xblock_settings[setting]
+        return result
+
+
+class AuthXBlockMixin(SettingsMixin, ProjectAPIXBlockMixin):
+
+    _DEFAULT_TA_ROLE = ('assistant',)
+
+    _SEE_DASHBOARD_ROLE_PERMS = "see_dashboard_role_perms"
```

Comment:
@e-kolpakov Yes that is true. Maybe change this to a ``access_dashboard_groups``, to keep it in-line with the api endpoint used to these "roles", just for sake of hypothetical people trying to merge it back to edX main repository. EdX api has an ``role`` endpoint and we don't use this endpoint to check if user has access the dashboard, and this would be misleading (unless EdX people often use groups as instance-wide roles, let me know if that is the case). 

If this argument doesn't convince you I'll change it :)